egs/wsj/s5/steps/cleanup/internal/get_pron_stats.py

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/env python
+
+# Copyright 2016  Xiaohui Zhang
+# Apache 2.0.
+
+from __future__ import print_function
+import argparse
+import sys
+import warnings
+
+# Collect pronounciation stats from a ctm_prons.txt file of the form output
+# by steps/cleanup/debug_lexicon.sh.  This input file has lines of the form:
+#  utt_id word phone1 phone2 .. phoneN
+#  e.g.
+#  foo-bar123-342  hello h eh l l ow
+# (and this script does require that lines from the same utterance be ordered in
+# order of time).
+# The output of this program is word pronunciation stats of the form:
+#  count word phone1 .. phoneN
+#  e.g.:
+#  24.0  hello h ax l l ow
+# This program uses various heuristics to account for the fact that in the input ctm_prons.txt
+# file may not always be well aligned.  As a result of some of these heuristics the counts will
+# not always be integers.
+
+def GetArgs():
+    parser = argparse.ArgumentParser(description = "Accumulate pronounciation statistics from "
+                                     "a ctm_prons.txt file.",
+                                     epilog = "See steps/cleanup/debug_lexicon.sh for example")
+    parser.add_argument("ctm_prons_file", metavar = "<ctm-prons-file>", type = str,
+                        help = "File containing word-pronounciation alignments obtained from a ctm file; "
+                        "each line must be <utt_id> <word> <phones>")
+    parser.add_argument("silence_file", metavar = "<silphone-file>", type = str,
+                        help = "File containing a list of silence phones.")
+    parser.add_argument("optional_silence_file", metavar = "<optional_silence>", type = str,
+                        help = "File containing the optional silence phone. We'll be replacing empty prons by this.")
```

Comment:
The overall script doesn't output any pron containing silence phones. I printed out those prons in the log file when running select_prons_bayesian.py:
  print('WARNING: removing the candidate pronunciation from phonetic-decoding: {0}:'
    '{1} whose soft-count from lattice-alignment is {2} cause it contains at'
   'least one silence phone.'.format(word, pron, count))